SubsoniciousKit/SubsoniciousKit/Network/Request/NetworkRequest.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  NetworkRequest.swift
+//  SubsoniciousKit
+//
+//  Created by Bilal on 05/07/2020.
+//  Copyright © 2020 Bilal Benlarbi. All rights reserved.
+//
+
+import CryptoKit
+import Foundation
+
+class NetworkRequest {
```

Comment:
because URLRequest is a struct and I'd like the Requests to inherit from NetworkRequest :)

Reply:
yep I saw this when a tested things, but maybe a class is not needed here, maybe you just want a func that transform a server into a URLRequest :P

Reply:
As all the web services require all the (authentication related) parameters defined by the NetworkRequest, I thought inheritance was a good candidate to fulfil that need. What do you think?

Reply:
In the review process I didn't see this conversation... weird.... 

For me you don't need inheritance to pass the authentications parameters.

You could have a struct that contains these parameters, and apply it to the URLRequest, this struct could live in the service class to shared with all your subclasses of services. And if you need to change it for only a small subset of services, you can just update the property.
If I was a request, I would prefer to live with a little structure that I can apply on myself, instead of being myself different because of that 😂 I don't know if that's relevant since I am not a request 😂

Reply:
With the inheritance approach we can override the path, the parameters (and even more if we like in the future). In my opinion, this approach is not so bad, so I'll keep it for now. Thanks a lot for the discussion :)